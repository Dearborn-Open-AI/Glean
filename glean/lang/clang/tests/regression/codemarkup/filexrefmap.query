query: cxx1.FileXRefMap _
transform: [gensort, []]
