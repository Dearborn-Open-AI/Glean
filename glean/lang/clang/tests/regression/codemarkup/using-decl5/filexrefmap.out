[
  "@generated",
  {
    "key": {
      "file": { "key": "test.cpp" },
      "fixed": [
        {
          "target": {
            "declaration": {
              "namespace_": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "N3" },
                      "parent": {
                        "key": {
                          "name": { "key": "N2" },
                          "parent": { "key": { "name": { "key": "N1" } } }
                        }
                      }
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 13,
                    "columnBegin": 1,
                    "lineEnd": 13,
                    "columnEnd": 31
                  }
                }
              }
            }
          },
          "ranges": [
            { "offset": 290, "length": 2 },
            { "offset": 241, "length": 2 }
          ]
        },
        {
          "target": {
            "declaration": {
              "namespace_": {
                "key": {
                  "name": { "key": { "name": { "key": "N1" } } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 9,
                    "columnBegin": 1,
                    "lineEnd": 29,
                    "columnEnd": 1
                  }
                }
              }
            }
          },
          "ranges": [
            { "offset": 523, "length": 2 },
            { "offset": 149, "length": 2 },
            { "offset": 52, "length": 2 }
          ]
        },
        {
          "target": {
            "declaration": {
              "namespace_": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "N2" },
                      "parent": { "key": { "name": { "key": "N1" } } }
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 11,
                    "columnBegin": 1,
                    "lineEnd": 27,
                    "columnEnd": 1
                  }
                }
              }
            }
          },
          "ranges": [
            { "offset": 527, "length": 2 },
            { "offset": 149, "length": 2 },
            { "offset": 52, "length": 2 }
          ]
        },
        {
          "target": {
            "declaration": {
              "function_": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": { "name": { "key": "id" } } },
                      "scope": { "global_": { } }
                    }
                  },
                  "signature": {
                    "key": {
                      "returns": { "key": "N1::N2::N3::Foo" },
                      "parameters": [
                        {
                          "name": { "key": "x" },
                          "type": { "key": "N1::N2::N3::Foo" }
                        }
                      ]
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 33,
                    "columnBegin": 1,
                    "lineEnd": 33,
                    "columnEnd": 13
                  }
                }
              }
            }
          },
          "ranges": [
            { "offset": 711, "length": 2 },
            { "offset": 57, "length": 2 }
          ]
        },
        {
          "target": {
            "declaration": {
              "record_": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "Bar" },
                      "scope": {
                        "namespace_": {
                          "key": {
                            "name": { "key": "N2" },
                            "parent": { "key": { "name": { "key": "N1" } } }
                          }
                        }
                      }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 22,
                    "columnBegin": 1,
                    "lineEnd": 25,
                    "columnEnd": 1
                  }
                }
              }
            }
          },
          "ranges": [ { "offset": 732, "length": 3 } ]
        }
      ],
      "variable": [
        [
          { "offset": 409, "length": 3 },
          { "offset": 10, "length": 3 },
          { "offset": 8, "length": 3 },
          { "offset": 54, "length": 3 },
          { "offset": 10, "length": 3 },
          { "offset": 8, "length": 3 },
          { "offset": 185, "length": 3 },
          { "offset": 57, "length": 3 }
        ],
        [
          { "offset": 541, "length": 3 },
          { "offset": 7, "length": 3 },
          { "offset": 118, "length": 3 },
          { "offset": 26, "length": 3 },
          { "offset": 26, "length": 3 },
          { "offset": 31, "length": 3 }
        ]
      ]
    }
  }
]